cluster-autoscaler/processors/nodegroupset/balancing_processor.go

```diff
@@ -103,7 +103,9 @@ func (b *BalancingNodeGroupSetProcessor) BalanceScaleUpBetweenGroups(context *co
 			NewSize:     currentSize,
 			MaxSize:     maxSize}
 		scaleUpInfos = append(scaleUpInfos, info)
-		totalCapacity += maxSize - currentSize
+		if maxSize-currentSize > 0 {
```

Comment:
nit: maybe we can also change the check above (in line 96) to "currentSize >= maxSize"?